examples/ecs/system_closure.rs

```diff
@@ -0,0 +1,42 @@
+use bevy::{log::LogPlugin, prelude::*};
+
+fn main() {
+    // create a simple closure.
+    let simple_closure = || {
+        info!("Hello from a simple closure!");
+    };
+
+    // create a closure, with a Local value to be initialized.
+    let complex_closure = |value: String| {
+        move |mut arg: Local<String>| {
+            *arg = value.clone();
+            info!("Hello from a complex closure! {:?}", arg);
+
+            // 'arg' will be saved between calls.
+            // you could also use an outside variable like presented in then inlined closures
+        }
+    };
```

Comment:
This is probably not working the way you want it to. Having a `Local` in a capturing closure is useless. Here you're cloning `value` to the local at every system execution, so whatever value would the local have would always be replaced by a fresh clone from `value`. arg would indeed be saved between calls, but the value wouldn't actually be available to the next call.

Instead you can directly use `value`. The outside closure put it in current scope as mutable, the inner closure capture it and actually modify it as every system execution.

```suggestion
    let complex_closure = |mut value: String| {
        move || {
            info!("Hello from a complex closure! {:?}", value);

            value = format!("{}o", value);
        }
    };
```

This could probably use more comments to explain what's happening...

Reply:
With some explanation, It makes sense. I should have tested this more aggressively. oops